### PR TITLE
Structures Dockerfile based on operation, and simplifies dev environment

### LINKED ISSUE #2973: Starting the dev service starts the db container which is nonessential
## Description
* Start the dev service with the command:
  ``` bash
  docker compose -f docker-compose.yml -f docker-compose.dev.yml up dev-service
  [+] Running 3/2
   ✔ Container mathesar_db           Created       0.1s 
   ✔ Container mathesar_dev_db       Created       0.0s 
   ✔ Container mathesar_service_dev  Created       0.0s 
  ```
* Notice that the `mathesar_dev_db` and `mathesar_service_dev` containers start as expected.
* Notice that an additional container `mathesar_db` also starts. This container is unused.

## Expected behavior
* `mathesar_db` container should not run.

## Notes
* I'm able to reproduce this consistently on my machine.
  * OS: macOS Big Sir
  * Chipset: Apple M1
* Anish has also reported that this issue is reproducible on his machine, which isn't a Mac.
* Mukesh and Brent have reported that this issue isn't reproducible on their environments.
* [Related Matrix discussions](https://matrix.to/#/!dwNKVEeYPiAqvkytob:matrix.mathesar.org/$Q-D7I7jacokt_xfcGqEnlWZjzTOC2e5SYwli_cSG8hQ?via=matrix.mathesar.org&via=matrix.org)

### PR DESCRIPTION
**Pull request by pavish**
Fixes #2973 

**Technical details**
* Dockerfile:
  * Adds structure to our Dockerfile, based on operation performed. Previously:
    * It wasn't clear where to add a new dependency that I wanted installed
  * Removes dockerize.It isn't needed after the changes to the compose file (see next parah)
  * Installs `gettext`, which is required for i18n project
* docker-compose.dev.yml
  * Removes dependency between `dev-service` and `service`, which fixes #2973 
  * Checks healthcheck of db, instead of using dockerize to regularly check if tcp port is open

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by pavish** [comment]
@mathemancer I'm just noticing that changes to the Dockerfile, and compose script, do not run our tests. We should probably change this behaviour to run tests.

**Review by mathemancer — CHANGES_REQUESTED** [review]
Looks good overall, but see my specific comment on the Dockerfile.

File: Dockerfile [hunk 1]
```diff
@@ -6,29 +6,44 @@ ENV NODE_MAJOR 18
...
+    echo "deb http://apt.postgresql.org/pub/repos/apt/ buster-pgdg main" > /etc/apt/sources.list.d/pgdg.list;
+
+# Add Node.js source
+RUN curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg; \
+    echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list;
+
+RUN apt-get update -y;
+
+# Install common dependencies
+RUN apt-get install -y --no-install-recommends sudo ca-certificates curl gnupg gettext nodejs locales;
```
[the remark this answers is not in the document]
Comment (pavish) [reply Dockerfile]: Thanks! I've added https://github.com/centerofci/mathesar/pull/3295/commits/f8011db570393363955bf29c4e19e31471eaa2c4 to resolve this.

**Review by mathemancer — APPROVED** [review]
Okay, this looks good to me.

**Pull request merged**